internal/jira/issue/issue.go

```diff
@@ -269,6 +269,11 @@ func CreateIssue(cfg *config.Config, issue *gogh.Issue, ghClient github.Client,
 		return fmt.Errorf("creating Jira issue: %w", err)
 	}
 
+	// in dry run mode we don't actually create the Jira issue so we shouldn't validate it
+	if cfg.IsDryRun() {
+		return nil
+	}
```

Comment:
Was this tested?
If I'm reading the current order correctly, soon after the error triggers, it is returned, which means this code path will never be checked in dry-run mode.

What I'd suggest instead is either (preferred):
- see if this is typed error in the upstream package
- if not, consider a custom error type like `errIssueCreateFailure`
- check if `err` is the same error type as `errIssueCreateFailure`

i.e., you want the error to occur and you want to ensure that the error that has occurred is the error type you're expecting

OR (less good):

```
	newIssue, err := jClient.CreateIssue(jIssue)
	if err != nil {
	        if cfg.IsDryRun() {
                        log.Info("Issues cannot be created in dry-run mode")
		        return nil
	        }

		return fmt.Errorf("creating Jira issue: %w", err)
	}
```

i.e., you want the dry-run check to be nested in the error check so that the code path is actually traversed

Reply:
Yes, it was tested. The CreateIssue function doesn't return an error, it works fine, but in dry run mode it doesn't create the issue.

The problem is here, in the next line: https://github.com/uwu-tools/gh-jira-issue-sync/blob/bcfad6fb339487b100a9818be03ab9ea54311e0d/internal/jira/issue/issue.go#L277-L279

Because once we create the issue we try to validate it by getting it from Jira. Which returns an error in dry run mode. That's why we should return before this and don't validate in dry run mode.

Reply:
Log line from the issue ticket:
> Error creating issue for #6. Error: getting Jira issue ......

Reply:
Thanks for the clarification, @rappizs!